### ISSUE TITLE
Allow "Reset" for each changed value in the Expression Profile and Individual BBs

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/MoBi#1560]
The user can overwrite values in the Individuals and Expression Profiles BBs. It should be possible to "reset" these values (so have a list tracking the changes...)

As @rwmcintosh pointed out here https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/2310#issuecomment-2338630226 a change can already be reversed in the history, but this is only feasible when the user wants to revert directly after changing the value.

**Comment by PavelBal** [comment]
Only store and allow rollback to the original value.

**Comment by rwmcintosh** [comment]
![image](https://github.com/user-attachments/assets/44a30de0-13e7-4d6d-abcb-5ae911a5ddf3)

Like in simulations

**Comment by PavelBal** [comment]
In following cases, a "reset" must be offered:

- Change of a value of a constant parameter (e.g., in an Individual: `Organism|Age` change from 30 to 32. "Reset" sets the value back to 30)
- Change of a unit of a constant parameter (e.g., in an Individual: `Organism|Age` change from 30 year(s) to 30 week(s). "Reset" sets the value back to 30 year(s))
- Change of a value of a constant parameter to a formula (either from a present or a new formula) (e.g., in an Individual: `Organism|Age` change from 30 to any formula. "Reset" sets the value back to the constant 30)
- Change of a value of a formula parameter to another formula (e.g., in an Individual: `Organism|BMI`. "Reset" sets the value back to the original formula). IS IT POSSIBLE?
- Change of a value of a formula parameter to a constant formula (e.g., in an Individual: `Organism|BMI`. "Reset" sets the value back to the original formula).
- Change a value from a distributed parameter (e.g., `Organism|Brain|Specific blood flow rate`) to a constant value. Reset returns to the distributed parameter
- Change a value from a distributed parameter (e.g., `Organism|Brain|Specific blood flow rate`) to a formula. Reset returns to the distributed parameter.

**Comment by msevestre** [comment]
agreed

**Comment by rwmcintosh** [comment]
I don't think this is a minor release issue. It will make 12.2 incompatible with 12.1

**Comment by rwmcintosh** [comment]
When exporting a snapshot, make sure the changed values are exported with the snapshot. PK-Sim will rebuild the original BB, then the changes can be applied.

**Comment by rwmcintosh** [comment]
Not sure about handling the distributed parameters. Those are different from the standard formula/value ones. When we update a distributed parameter, we first convert it to a constant and then remove all the sub-parameters that describe the distribution. Do you want to allow the user to reset the value to the initial constant that was created after it was converted from distribution?

**Comment by PavelBal** [comment]
So I understand that 

> Change a value from a distributed parameter (e.g., Organism|Brain|Specific blood flow rate) to a constant value. Reset returns to the distributed parameter

should not be possible. Okay for now, so "reset" reverts to the constant value. We should create a separate issue for "allow reset to distributed parameter" for later.

**Comment by rwmcintosh** [comment]
Sure. Created https://github.com/Open-Systems-Pharmacology/MoBi/issues/2211

Let's keep in in v13 for possible implementation still.

**Comment by rwmcintosh** [comment]
> * Change a value from a distributed parameter (e.g., `Organism|Brain|Specific blood flow rate`) to a formula. Reset returns to the distributed parameter.

I'd rather disable that option #2212

**Comment by PavelBal** [comment]
Okay.
